streams/src/main/java/org/apache/kafka/streams/processor/internals/SinkNode.java

```diff
@@ -57,7 +57,7 @@ public void init(ProcessorContext context) {
     public void process(K key, V value) {
         // send to all the registered topics
         RecordCollector collector = ((RecordCollector.Supplier) context).recordCollector();
-        collector.send(new ProducerRecord<>(topic, key, value), keySerializer, valSerializer, partitioner);
+        collector.send(new ProducerRecord<>(topic, null, context.timestamp(), key, value), keySerializer, valSerializer, partitioner);
```

Comment:
It is certainly possible to add ```ProducerRecord(String, Long, K, V)```. How about ```ProducerRecord(String, Long, V)```? It is not possible due to ambiguity. It conflicts with ```ProducerRecord(String, K, V)```. So, we cannot define constructors with all possible combinations, and we don't have to. I am not sure how much value a new constructor adds. I am reluctant to add a new constructor in this PR.